src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
we dont want to get machine producers of type 'Number' for example
```suggestion
public interface MachineProducer<T extends Machine> {
```